### DISCUSSION TITLE
Interop.Errors.cs

### DISCUSSION THREAD

**Discussion by olapchuk** [dotnet/runtime#40319] [General] [answered]
Is there any public interface to this stuff
src/libraries/Common/src/Interop/Unix/Interop.Errors.cs
Or how do I convert error code (errno comes from native library) into correct .net exception ?

**Comment by danmoseley** [comment, marked as answer]
`Win32Exception()` works on Unix (it's a terrible name - historical) and you can either pass the errno or it will get it itself with `Marshal.GetLastWin32Error()` (again, historical name) and then it will use the code to get a message using `strerror` on Unix.

https://github.com/dotnet/runtime/blob/c87e75e8ef260ccc6e979f9df1cc012536ee036f/src/libraries/Microsoft.Win32.Primitives/src/System/ComponentModel/Win32Exception.cs#L31
https://github.com/dotnet/runtime/blob/c87e75e8ef260ccc6e979f9df1cc012536ee036f/src/libraries/Common/src/Interop/Unix/Interop.Errors.cs#L145

Does that help?

**Reply by olapchuk** [reply]
Thanks, that helps
And is there sometihng like PosixException/GetPosixErrorMessage for windows environment ?

**Reply by danmoseley** [reply]
It's Win32Exception on both
